File: app/src/main/java/com/johnguant/redditthing/auth/RedditAccountService.java
package com.johnguant.redditthing.auth;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;

public class RedditAccountService extends Service{

    @Override
    public void onCreate(){
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        RedditAuthenticator authenticator = new RedditAuthenticator(this);
        return authenticator.getIBinder();
    }
}

File: app/src/main/java/com/johnguant/redditthing/auth/RedditAccountService.kt
package com.johnguant.redditthing.auth

import android.app.Service
import android.content.Intent
import android.os.IBinder

class RedditAccountService : Service() {

    override fun onCreate() {}

    override fun onBind(intent: Intent): IBinder? {
        val authenticator = RedditAuthenticator(this)
        return authenticator.iBinder
    }
}
